File: benchmarks/c/bench_kilo_micro_substring_views_only.c
#include <stdint.h>
#include <string.h>

int main(void) {
  const int64_t ops = 300000;
  const char text[] = "line-seed-abcdef";
  const int len = (int)(sizeof(text) - 1);
  const int split = len / 2;
  char left[32];
  char right[32];

  for (int64_t i = 0; i < ops; i++) {
    memcpy(left, text, (size_t)split);
    left[split] = '\0';
    memcpy(right, text + split, (size_t)(len - split));
    right[len - split] = '\0';
  }

  return (int)(((int64_t)(unsigned char)left[0] + (int64_t)(unsigned char)right[0] + len) & 0xFF);
}

File: benchmarks/rust/bench_kilo_micro_substring_views_only.rs
use std::hint::black_box;

fn main() {
    let ops = black_box(300_000usize);
    let text = black_box("line-seed-abcdef");
    let len = black_box(text.len());
    let split = black_box(len / 2);
    let mut left = black_box(text);
    let mut right = black_box(text);

    for _ in 0..ops {
        left = black_box(&text[..split]);
        right = black_box(&text[split..]);
    }

    let checksum = black_box(left.len() + right.len() + len);
    std::process::exit((checksum & 0xff) as i32);
}
